web/utils.py

```
@@ -414,6 +416,407 @@ def create_leaderboard_context(
+    Returns:
+        stripe.Subscription: The updated subscription
+    """
+    return stripe.Subscription.modify(
+        subscription_id,
+        items=[{"price": price_id}],
+        payment_behavior="allow_incomplete",
+        proration_behavior="create_prorations",
+    )
+
+
+def create_subscription(user: "User", plan_id: int, payment_method_id: str, billing_period: str) -> dict[str, Any]:
```

Ruff (0.8.2)

`create_subscription` is too complex (13 > 10)

(C901)